src/editor/Gui.cpp

```diff
@@ -222,6 +229,66 @@ void Gui::pasteLightmap() {
 	app->mapRenderers[app->pickInfo.mapIdx]->reloadLightmaps();
 }
 
+
+void ExportModel(Bsp* map, int id)
+{
+	map->update_ent_lump();
+
+	Bsp* tmpMap = new Bsp(map->path);
+
+	BSPMODEL tmpModel = map->models[id];
+	
+	//for (int i = 0; i < tmpMap->leafCount; i++)
+	//{
+	//	BSPLEAF& tmpLeaf = tmpMap->leaves[i];
+	//	tmpLeaf.nVisOffset = -1;
+	//	for (int n = 0; n < 3; n++)
+	//	{
+	//		tmpLeaf.nMins[n] = tmpModel.nMins[n];
+	//		tmpLeaf.nMaxs[n] = tmpModel.nMaxs[n];
+	//	}
+	//}
+
+	Entity* tmpEnt = new Entity(*map->ents[0]);
+
+	vec3 origin = vec3(0,0,0);
+	for (int i = 0; i < map->ents.size(); i++)
+	{
+		if (map->ents[i]->getBspModelIdx() == id)
+		{
+			origin = map->ents[i]->getOrigin();
+			logf("Save model origin: %f %f %f to worldpsawn entity\n", map->ents[i]->getOrigin().x, map->ents[i]->getOrigin().y, map->ents[i]->getOrigin().z);
+		}
+	}
+
+	for (int i = 1; i < tmpMap->ents.size(); i++)
+	{
+		delete tmpMap->ents[i];
+	}
+
+	tmpMap->ents.clear();
+	if (origin != vec3(0,0,0))
+		tmpEnt->setOrAddKeyvalue("origin", origin.toKeyvalueString());
+	tmpMap->ents.push_back(tmpEnt);
+	tmpMap->update_ent_lump();
+
+	tmpMap->modelCount = 1;
+	tmpMap->models[0] = tmpModel;

```

Comment:
I think this what is complicating the merge/import. The entity model is being exported as the worldspawn model here. The merger will try to combine this with the other map's worldspawn model, instead of appending to the list of models. Try deleting most of the worldspawn model structures and exporting the entity model as an entity model. You should be able to merge the maps without any code changes to BspMerger. After the maps are merged, then you can spawn an entity using the newly imported model index.

I think that would be cleaner than having the separate `MergeSecondsMapAsModel` mode in the merger. There might still need to be code to skip merging model 0, but it would make more sense.

Reply:
I got really works models at outpup (is valid and can be used in game). I removed MergeSecondsMapAsModel mode because it not works. )

Reply:
I tried also to add empty BSP model as worldspawn.  I see changes in bspguy, but in game I can't see these models. Now this option removed from bspguy.